### ISSUE TITLE
boxmode 'group' on 'logscale' boxgap issue

### ISSUE THREAD

**Issue by jeonsimon** [plotly/plotly.py#3970]
Hi, There is an issue on boxplot.

when I use 
fig.update_layout(
    boxmode = 'group',
    xaxis=dict(type="log")
)

As X value increases, boxplot gap decreases more and more.
I want to same gap whatever value chagned.

**Comment by alexcjohnson** [comment]
Yes, makes sense that if we’re going to treat individual boxes that way on log axes we should do the same for grouped boxes. And violins too. And bars? That might be a bigger change than we can make to the default behavior without considering it breaking, ideally we’d make an option to scale either way. Moving to the plotly.js repo.
